pkg/operator/csi/controller/credentials/controller.go

```diff
@@ -0,0 +1,91 @@
+package credentials
+
+import (
+	"context"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	opv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// Controller is a simple controller that maintains a CredentialsRequest static manifest.
+// It uses unstructured.Unstructured as currently there's no API type for this resource.
+type Controller struct {
+	name            string
+	operatorClient  v1helpers.OperatorClient
+	targetNamespace string
+	manifest        []byte
+	dynamicClient   dynamic.Interface
+}
+
+// New returns a Controller.
+func New(
```

Comment:
This is a good controller, but I think our overall experience would be better if your read and apply functions were in resourceread and resourceapply, and then we simply make https://github.com/openshift/library-go/blob/master/pkg/operator/staticresourcecontroller/static_resource_controller.go#L61  compatible.

Is there a reason this is impractical?  If so, simply moving the resourceread and resourceapply functions so we can later add it is ok with me, but I'd like a TODO describing the limitations here.

Reply:
Move the read/apply functions to resourceread and resourceapply.

I started adding code to handle the CredentialsRequest in the generic apply function but that's going to take more time that I anticipated. I'll work on that (and remove this controller) in a subsequent PR.